src/lib/plot/hint.js

```diff
@@ -199,17 +285,18 @@ class Hint extends PureRenderComponent {
   _getPositionInfo() {
     const {
       value,
-      orientation: initialOrientation} = this.props;
+      horizontalAlign,
+      verticalAlign
+    } = this.props;
 
     const x = getAttributeFunctor(this.props, 'x')(value);
     const y = getAttributeFunctor(this.props, 'y')(value);
 
-    const orientation = initialOrientation === ORIENTATION_AUTO ?
-      this._getOrientationFromAuto(x, y) : initialOrientation;
+    const align = this._getAlign(x, y);
 
     return {
-      style: this._getOrientationStyle(orientation, x, y),
-      className: this._getOrientationClassName(orientation)
+      style: this._getAlignStyle(align, x, y),
+      className: this._getAlignClassNames(align)
```

Comment:
Can be `this.getClassNames(this.props)`, and that method can call `_getOrientationClassName` or `_getAlignClassNames` depending on whether `align` is used.